solr/core/src/java/org/apache/solr/update/processor/ClassificationUpdateProcessorFactory.java

```diff
@@ -92,8 +92,7 @@ public void init(final NamedList<?> args) {
       String algorithmString = params.get(ALGORITHM_PARAM);
       Algorithm classificationAlgorithm;
       try {
-        if (algorithmString == null
-            || Algorithm.valueOf(algorithmString.toUpperCase(Locale.ROOT)) == null) {
```

Comment:
This doesn't look related to version catalogs?

Reply:
We need to be careful with errorprone updates. See https://github.com/apache/solr/blob/main/gradle/validation/error-prone.gradle#L68 we want to make sure we list and explicitly enable/disable checks. Just upgrading means we aren't going to handle all the checks we may not want to run since they are either covered by other tools (ie: forbidden apis) or they are not necessary for our codebase.

Reply:
I agree that we should not take error-prone updates lightly. But we still have to update and sync error-prone versions one way or another.

Would it suffice in your opinion if we update error-prone to 2.28.0 in a separate PR first before proceeding here?

Reply:
I agree that should be done first so we can make sure that all the rules are put into https://github.com/apache/solr/blob/main/gradle/validation/error-prone.gradle#L68

Reply:
With the changes from https://github.com/apache/solr/pull/2862, this should no longer be an issue in this PR. As mentioned in the PR, further rule evaluation is necessary to enable a few more useful rules for the future.